src/isl/isl_constraint.rs

```diff
@@ -99,6 +111,12 @@ impl IslConstraint {
                 )?;
                 Ok(IslConstraint::AnyOf(types))
             }
+            "byte_length" => Ok(IslConstraint::ByteLength(Range::from_ion_element(
+                value, true, // Pass true as byte_length will have non negative range
```

Comment:
It may be worth defining an `enum` with variant names that communicate the intent of this `bool` so you don't have to explain each usage with a comment.